### PR TITLE
add Forward Error Correction (FEC) documentation

### PR DESCRIPTION
**Pull request by KanjiMonster**
Add documentation about configuring FEC, and switch and software limitations.

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>

### CODE REVIEW

File: platform_configuration/forward_error_correction.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+title: Forward Error Correction (FEC)
+parent: Platform Configuration
+nav_order: 3
+---
+
+### Enabling Forward Error Correction (FEC) on a port
+
+By default Broadcom switches do not enable any Forward Error Correction on ports.
+
+To enable Forward Error Correction use the client_drivshell tool.
```
Comment (hilmarm) [inline platform_configuration/forward_error_correction.md:11]:
```suggestion
To enable Forward Error Correction use the [client_drivshell tool.](/tools/ofdpa_client_tools.html#client_drivshell)
```

Comment (hilmarm) [reply platform_configuration/forward_error_correction.md:11]: Nice to have

File: platform_configuration/forward_error_correction.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+There are multiple algorithms supported:
+
+* Base-R, defined in IEEE 802.3, clause 74.
+* Reed-Solomon (RS) for 100G ports, defined in IEEE 802.3, clause 91.
+* Reed-Solomon (RS) for 25G ports, defined in IEEE 802.3 clause 108.
+
+The configuration uses the clause numbers to references to the different modes.
```
Comment (hilmarm) [inline platform_configuration/forward_error_correction.md:19]:
```suggestion
The configuration uses the clause numbers as references to the different modes.
```

File: platform_configuration/forward_error_correction.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+To e.g. enable RS for 100G on the first port, simple run:
```
Comment (hilmarm) [inline platform_configuration/forward_error_correction.md:21]:
```suggestion
E.g., to enable RS for 100G on the first port, simply run:
```

File: platform_configuration/forward_error_correction.md (hunk 2 above)
Comment (KanjiMonster) [reply platform_configuration/forward_error_correction.md:19]: Actually what I wanted was
> The configuration uses the clause numbers to reference the different modes.

**Comment by hilmarm** [comment]
Well written, added a couple of grammatical suggestions (sorry for not batching them) and added a link to client_drivshell.
The rest lgtm

File: platform_configuration/forward_error_correction.md (hunk 2 above)
Comment (hilmarm) [reply platform_configuration/forward_error_correction.md:19]: Yes exatcly, that would have been my second suggstion. Sounds better.

**Comment by KanjiMonster** [comment]
Applied the suggestions.

**Review by jklare — APPROVED** [review]
lgtm

**Pull request merged**